### ISSUE TITLE
Torch Script mixed with source code

### ISSUE THREAD

**Issue by FynnBe** [bioimage-io/spec-bioimage-io#64]
@ahanio is in the process of preparing his pytorch model https://github.com/Ahanio/nucleoli_segment for our zoo. He encountered the following scenario:
- saving the weights as pytorch state dict does not work (probably due to inferno version clash)
- saving the weights as pytorch script does avoid the first issue, however the custom numpy pre/postprocessing is incompatible with torch script. 
1. **Source with custom weights?** A model (just a python class) with a method 'load_custom_weights(file_like)' to control given weights format (e.g. 'custom_weights'; could internally use torch script in this example)
2. **explicit format 'wrapped_[torch_script/keras_hdf5...]'?** A model (just a python class) with a 'model' property that is set from another weights format.

Baseline: A model with custom pre/postprocessing might rely on source code, while standard weight formats can eleviate version clashes. 'solution' 1. or 2. are certainly not the only approaches to this problem. Let's discuss!

For the time being @ahanio is looking into porting his model weights to be compatible with a newer inferno version, so that we might use the regular 'pytorch_state_dict' for his model.

**Comment by Ahanio** [comment]
@FynnBe thanks for the problem description.
Installing inferno via conda solved the issue. But I met another problem.
When I do ''' model = torch.load('/whole/model/path') ''' everything works fine.
But if I try that: 
''' model = ModelDefinition()
    model.load_state_dict('/weights/path') '''
The predictions gets damaged. 

I tried loading the model the first way, saving just state dict and running the second method (result is the same).
Downgrading inferno and torch versions (to the once used in model training) doesn't help.

Here is the comparison of predictions
![test1](https://user-images.githubusercontent.com/14354055/111533695-21edf400-8778-11eb-8361-42c19093d1fd.png)
![test2](https://user-images.githubusercontent.com/14354055/111533699-231f2100-8778-11eb-9e49-8429ed66c1e4.png)

**Comment by FynnBe** [comment]
This is very hard to give a remote diagnosis on... 
To debug this I'd suggest:
1. compare `model.state_dict()` and `model.load_state_dict('/weights/path').state_dict()`.
2. Check that your model and it's submodules are instantiated correctly: https://pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module
3. compare subparts of the two models for differences.

**Comment by constantinpape** [comment]
Is this maybe a problem with normalization not being applied properly? The predictions look a bit like that.

**Comment by Ahanio** [comment]
Found it: there was an issue with batch_norm layer in state_dict(). Now fixed it. 
Thank you for the help!
